rdmo/questions/static/questions/js/catalogs.js

```diff
@@ -94,19 +95,50 @@ angular.module('catalogs', ['core'])
         service.showQuestionSets = !(sessionStorage.getItem('options_showQuestionSets') === 'false');
         service.showQuestions = !(sessionStorage.getItem('options_showOptions') === 'false');
 
-        resources.catalogs.query({list_action: 'index'}, function(response) {
+        if (angular.isUndefined(service.filterCatalogsCurrentSite)) {
+            if (sessionStorage.getItem('options_filterCatalogsCurrentSite') === null) {
+                // when user arrives on page for the first time, set filterCatalogsCurrentSite to true
+                service.filterCatalogsCurrentSite = true;
+            } else if (sessionStorage.getItem('options_filterCatalogsCurrentSite') === 'false') {
+                // when user switches language and options_filterCatalogsCurrentSite was false before
+                service.filterCatalogsCurrentSite = false;
+            } else if (sessionStorage.getItem('options_filterCatalogsCurrentSite') === 'true') {
+                // when user switches language and options_filterCatalogsCurrentSite was true before
+                service.filterCatalogsCurrentSite = true;
+            }
+        }
+
+        var catalogs_params = {
+            list_action: 'index'
+        }
+        if (service.filterCatalogsCurrentSite) {
+            // query only catalogs for this site
+            catalogs_params['sites'] = site_id
+        }
+
+        resources.catalogs.query(catalogs_params, function(response) {
             service.catalogs = response;
 
             // try to get the catalog from the address bar
             var catalog_id = $location.path().replace(/\//g,'');
 
+            // check if catalog_id is in the array of catalogs
             if (catalog_id) {
-                service.current_catalog_id = catalog_id;
-            } else if (service.catalogs.length) {
+                // check if catalog_id is in catalogs
+                // required for when user switches filterCatalogsCurrentSite
+                var catalog_id_isin_catalogs = $filter('filter')(service.catalogs, {'id': catalog_id});
```

Comment:
Here was `{'id': 1}` before, replacing by `{'id': catalog_id}`  is the only change I could find.